include/fastdds/rtps/common/CDRMessage_t.h

```diff
@@ -196,6 +205,8 @@ struct RTPS_DllAPI CDRMessage_t final
     uint32_t pos;
     //!Max size of the message.
     uint32_t max_size;
+    //!Size allocated on buffer. May be higher than max_size.
+    uint32_t reserved_size;
```

Comment:
I can only see `max_size` and `reserved_size` differing after `reserve()`, and in that case it seems that `max_size` can end up being larger than `reserverd_size`, not the contrary

Even when the `realloc` fails, we set the `max_size` to the attempted size leaving the `reserved_size` with the old (lower) value. 

What is the intention keeping a separate `max_size`?

Reply:
This is about calling `reserve()` first with a big size and then with a shorter size. In that case we dont't want the reallocation, but we want the shorter size being assigned to `max_size`

Reply:
It makes sense